donut/modules/calendar/static/script.js

```diff
@@ -0,0 +1,388 @@
+      // Half of the time it'll work and the other half wouldn't
+      timeout:3000, 
+      error: function(jqXHR, textStatus){
+        if(textStatus === 'timeout')
+        {     
+            // Access the backup
+            $.ajax({
+                url: '/1/calendar_events_backup',
+                type: 'POST',
+                data:{
+                    month:month+1,
+                    year:year
+                },
+                success:function(data){
+                    if (!window.parent.location.href.includes('sync')){
+                        $('#failover_message').show();
+                    }
+                    process_events(data, month, year); 
+                }
+            });
+        }
+      }
+    });
+
+  }
+}
+
+// Render Calendar
+function renderCalendar(startDay, totalDays, currentDate) {
+  var currentRow = 1;
+  var currentDay = startDay;
+  var $table = $('#cal');
+  var $week = getCalendarRow();
+  var $day;
+  var i = 1;
+  for (; i <= totalDays; i++) {
+    $day = $week.find('td').eq(currentDay);
+    $day.text(i);
+    if (i === currentDate) {
+      $day.addClass('today');
+    }
+
+    $day.append("</br>");
+    var wrapper = $('<div></div>').attr('id', 'day '+i).addClass('day');
+    $day.append(wrapper);
+    // +1 next day until Saturday (6), then reset to Sunday (0)
+    currentDay = ++currentDay % 7;
+
+    // Generate new row when day is Saturday, but only if there are
+    // additional days to render
+    if (currentDay === 0 && (i + 1 <= totalDays)) {
+      $week = getCalendarRow();
+      currentRow++;
+    }
+  }
+}
+
+// Clear generated calendar
+function clearCalendar() {
+  var $trs = $('tr').not(':eq(0)').not($('.calendar'));
+  $trs.remove();
+  $('#extraInfo').hide();
+  $('#extraInfo_triangle').hide();
+  $('.month-year').empty();
+}
+
+// Generates table row used when rendering Calendar
+function getCalendarRow() {
+  var $table = $('#cal');
+  var $tr = $('<tr/>');
+  for (var i = 0, len = 7; i < len; i++) {
+    $tr.append($('<td/>'));
+  }
+  $table.append($tr);
+  return $tr;
+}
+
+function myCalendar() {
+  var month = d.getMonth();
+  var day = d.getDay();
+  var year = d.getFullYear();
+  var date = d.getDate();
+  var totalDaysOfMonth = daysOfMonth[month];
+  var counter = 1;
+
+  var $ym = $('<h3>');
+
+  $ym.text(content[month] + ' ' + year);
+  $ym.appendTo('.month-year');
+
+  var dateToHighlight = 0;
+
+  // Determine if Month && Year are current for Date Highlight
+  if (CURRENT_DATE.getMonth() === month && CURRENT_DATE.getFullYear() === year) {
+    dateToHighlight = date;
+  }
+
+  //Getting February Days Including The Leap Year
+  if (month === 1) {
+    if ((year % 100 !== 0) && (year % 4 === 0) || (year % 400 === 0)) {
+      totalDaysOfMonth = 29;
+    }
+  }
+
+  // Get Start Day
+  renderCalendar(getCalendarStart(day, date), totalDaysOfMonth, dateToHighlight);
+  if (curYear != year ||$.isEmptyObject(events))
```

Comment:
This check is also being performed in `getEvents()`. Would be good to only do it in 1 place